grudge/tools.py

```diff
@@ -235,4 +239,105 @@ def build_jacobian(actx, f, base_state, stepsize):
     return mat
 
 
+def map_subarrays(
+        f: Callable[[np.ndarray], np.ndarray],
```

Comment:
`f` can also map scalars to arrays or arrays to scalars; not sure how best to indicate this with type hints here.

Reply:
I think it just needs to be `Any`, since the scalar types can be arbitrary.